### ISSUE TITLE
Example project

### ISSUE THREAD

**Issue by geelen** [sullenor/css-modules-require-hook#13]
Much in the same spirit as the [browserify](https://github.com/css-modules/browserify-demo), [jspm](https://github.com/css-modules/jspm-demo) and [webpack](https://github.com/css-modules/webpack-demo) demos, I'd love to see a simple project using css-modules-require-hook to load CSS and render templates with it on the server side.

Is that something you could take on, @sullenor?

**Comment by geelen** [comment]
I'm thinking something simple like express, but honestly whatever you'd like to use.

**Comment by sullenor** [comment]
Sure

I started one earlier. I wanted to try react + redux, so I made a project-stub with a simple demo of the popular game. It uses express on the server side and still lacks css-modules (I will update it soon :)).
I'm not sure if it will be a good example. It can be a bit difficult for the beginners.

See the link below:
https://github.com/sullenor/skeleton

**Comment by sullenor** [comment]
@geelen Finally, I managed to make a simple project. In reference to the popular tradition I decided to make a todoApp :)
Figuring out how to use the new `redux-react` package took some time from me. Anyways, check it out!

https://github.com/sullenor/todos

**Comment by geelen** [comment]
That's great! I'm not quite clear on what's different between this and a normal React/Redux app though? Maybe flesh out the README to point folks at the important points?

**Comment by sullenor** [comment]
Yeah, my bad. I guess I was obsessed a little with React and Redux. May be I should make something simpler like browserify- or webpack- demos.

**Comment by geelen** [comment]
No I think what you've done is great, just maybe help some people understand how it all hangs together.

You could also do a few interesting things, like move [the global body styles](https://github.com/sullenor/todos/blob/master/components/app/index.css#L1) into an actual CSS module then reference that from the `index.jade` somehow? That should be possible, right?

**Comment by sullenor** [comment]
> You could also do a few interesting things, like move the global body styles into an actual CSS module then reference that from the index.jade somehow? That should be possible, right?

I guess so :) Honestly I wasn't sure in choosing a good variant to do that.
Also I think about opportunity of making flexible way to change themes with using CSS variables. Looks like adding another postcss-plugin will help here.

**Comment by sullenor** [comment]
Thank you for sharing your thoughts with me, I'll try to update it in next few days.

**Comment by boyswan** [comment]
A webpack demo would be great if possible, trying to figure out how to fix the path errors

**Comment by bdefore** [comment]
:+1: for a webpack example. i'm looking for a way to render serverside modules that use compose and as i understand it's still unsolved, although this library could conceivably fix it

**Comment by canvaskisa** [comment]
Hey, guys! I've created a simple webpack + redux + css-modules-require-hook todomvc example app, please check it out https://github.com/canvaskisa/redux-css-modules-todomvc
I'd very appreciate your comments and questions, hope that example will help.

**Comment by sullenor** [comment]
Okay, I made another https://github.com/css-modules/css-modules-require-hook/tree/master/demo and wrote detailed description. If you'll have questions, please, send me a note :)

**Comment by StevenIseki** [comment]
Thanks @sullenor for your awesome easy to follow demo. I made a similar [one](https://github.com/StevenIseki/react-router-redux-example) based on yours with react-router-redux and isomorphic-fetch. Great work :+1:

**Comment by max-gram** [comment]
I'm using `css-modules-require-hook` in [react-saga-universal](https://github.com/max-gram/react-saga-universal) boilerplate
